subprojects/core/src/main/java/org/gradle/api/internal/tasks/TaskStateInternal.java

```diff
@@ -20,13 +20,16 @@
 import org.gradle.api.internal.TaskOutputCachingState;
 import org.gradle.api.tasks.TaskState;
 
+import static org.gradle.api.internal.tasks.TaskExecutionOutcome.*;
+
 public class TaskStateInternal implements TaskState {
```

Comment:
I don't like adding the logic to compute how much a task saved to `TaskStateInternal`. We also need to add some more timers to calculate durations.
All this information is already present as `BuildOperation`s. Could we listen for those and update the time saved/spent accordingly? I guess we would need to add the original duration of the task either to `ExecuteTaskBuildOperationResult` or better to `UnpackOperationResult`.
Then we could have a build operation listener which does the counting, right? All the times should already be captured as part of build operations, shouldn't they? At least we already have the times in build scans, so I guess we know them.